### PR TITLE
Base CLI setup

### PR DESCRIPTION
**Pull request by zulcss**
This branch sets up the base CLI agent bits so we can work on top of it:

- Setup BaseCommand class which is basically the base script that was done in shell script.
- Add setup_logigng method so that we can log errors and communicate with the user.
- Remove unused code
- Use CliContext class to create a click object. We create a click object so we can save the state across different CLI commands.

### CODE REVIEW

File: lib/pbench/agent/base.py
Comment (dbutenhof) [inline lib/pbench/agent/base.py:101]: Shouldn't there be an `else` returning `path`??? Or maybe just remove the `else:` on the `if path:` and return the input `path` if it either was already a Path object, or was None?

File: lib/pbench/agent/fs.py
Comment (dbutenhof) [inline lib/pbench/agent/fs.py:18]: "ensures the directory is empty" ... ha. But I guess at the command level, deleting an empty directory and re-creating it shouldn't be a big deal.

File: lib/pbench/agent/utils.py
Comment (dbutenhof) [inline lib/pbench/agent/utils.py:24]: Same comments as before.

1. Although I suspect that fine tuning log levels is rarely all that useful, you're overriding that option by bundling them together.
2. You're checking `_PBENCH_TOOL_MEISTER_START_LOG_LEVEL` twice.

File: lib/pbench/cli/agent/__init__.py
Comment (dbutenhof) [inline lib/pbench/cli/agent/__init__.py:8]: "execution"

File: lib/pbench/cli/agent/__init__.py
Comment (dbutenhof) [inline lib/pbench/cli/agent/__init__.py:6]: An empty line between paragraphs would make it easier to read.

File: lib/pbench/agent/fs.py
Comment (zulcss) [reply lib/pbench/agent/fs.py:18]: Done

File: lib/pbench/cli/agent/__init__.py
Comment (zulcss) [reply lib/pbench/cli/agent/__init__.py:8]: Done

File: lib/pbench/cli/agent/__init__.py
Comment (zulcss) [reply lib/pbench/cli/agent/__init__.py:6]: Done

File: lib/pbench/agent/utils.py
Comment (zulcss) [reply lib/pbench/agent/utils.py]: Fixed

File: lib/pbench/agent/base.py
Comment (dbutenhof) [inline lib/pbench/agent/base.py:101]: But if `path` is `None`, `pathlib.PurePath` is going to be unhappy. Maybe that's OK if we know it can never be None; but you were checking explicitly before.

Comment (zulcss) [reply lib/pbench/agent/base.py]: Yeah fixed

**Review by dbutenhof — DISMISSED** [review]

File: lib/pbench/agent/base.py
Comment (portante) [inline lib/pbench/agent/base.py:33]: Can we make `safe_mkdir` ensure the directory is either created or it raises an exception?

And it looks like it already does raise an exception which does not appear to be caught here.

**Review by portante — CHANGES_REQUESTED** [review]
Can we address the exception handling in `safe_mkdir` and then rebase/squash to one commit?

File: lib/pbench/agent/base.py
Comment (zulcss) [reply lib/pbench/agent/base.py]: Done

**Review by portante — CHANGES_REQUESTED** [review]
Looks good!  Let's get this rebased and a second review so we can merge.

**Review by portante — CHANGES_REQUESTED** [review]
Can you squash to one commit?  Thanks!

**Review by portante — APPROVED** [review]
Ship it!

**Review by dbutenhof — APPROVED** [review]

**Pull request merged**
